src/main/java/org/opensearch/ad/feature/CompositeRetriever.java

```diff
@@ -342,7 +342,14 @@ Optional<CompositeAggregation> getComposite(SearchResponse response) {
          * @return true if the iteration has more pages.
          */
         public boolean hasNext() {
-            return (iterations == 0 || (totalResults > 0 && afterKey != null)) && expirationEpochMs > clock.millis();
+            long now = clock.millis();
+            if (expirationEpochMs <= now) {
+                LOG.info("Time is up, afterKey: " + afterKey + " " + expirationEpochMs + " " + now);
+            }
+            if ((iterations > 0 && afterKey == null) || totalResults > maxEntities) {
+                LOG.info("Finished in " + (now - startMs) + " msecs. ");
```

Comment:
nit - should these be debug level logging?

Reply:
yeah, changed